styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java

```diff
@@ -89,17 +80,26 @@ public void transitionInto(RunState state) {
             } else {
               LOG.error(msg, e);
             }
-            stateManager.receive(Event.runError(state.workflowInstance(), e.getMessage()), 
+            stateManager.receive(Event.runError(state.workflowInstance(), e.getMessage()),
                 state.counter() + 1);
           } catch (IsClosedException isClosedException) {
             LOG.warn("Failed to send 'runError' event", isClosedException);
           }
+          return;
+        }
+
+        final Event submitted = Event.submitted(state.workflowInstance(), runSpec.executionId());
+        try {
+          stateManager.receive(submitted, state.counter());
```

Comment:
Good catch, we should not use `state.counter() + 1` at all.

Added a comment on the reason for emitting `submitted` after starting the execution.

Reply:
> Emit submitted event first to guarantee it is observed before events from the pod

Isn't this still valid? If we have received events from pod, this `submitted` event will become invalid due to counter mismatching.

Reply:
The reason to emit the `submitted` event was eliminated two years ago: https://github.com/spotify/styx/pull/198

The original reason was so that events from k8s could observe the execution id that was created (and persisted) as part of the `submitted` event.

But now we're persisting the execution id in the `submit` event, so emitting `submitted` after starting the execution is safe.